packages/roosterjs-editor-plugins/lib/plugins/CustomReplace/CustomReplace.ts

```diff
@@ -98,6 +105,13 @@ export default class CustomReplacePlugin implements EditorPlugin {
             return;
         }
 
+        if (
+            replacement.shouldReplace &&
+            !replacement.shouldReplace(replacement, searcher.getWordBefore())
```

Comment:
You already have stringToSearch, no need to call getWordBefore() again

Reply:
I'm only replacing st, nd, rd, th, then I'm using getWordBefore() to find if there's a number before st, nd, rd, th.

Reply:
Then I' don't have to replace  from 1st to 9th.

Reply:
can you give me an example, when I type "1st", what are the values of stringToSearch and getWordBefore()?

Reply:
1. Type "1st "
2.  The stringToSearch will be "st " and the getWordBefore() will be "1st "
3. The callback will use getWordBefore() to check if there's a 1 before "st "
4. If true, replace st for 1<sup>st</sup>&nbsp;
5. If not, do not replace.